service/matching/pri_task_reader.go

```diff
@@ -173,9 +173,9 @@ func (tr *priTaskReader) completeTask(task *internalTask, res taskResponse) {
 		tr.SignalTaskLoading()
 	}
 
-	tr.lock.Unlock()
+	tr.backlogMgr.db.updateAckLevelAndCount(tr.subqueue, tr.ackLevel, -numAcked)
 
-	tr.backlogMgr.db.updateAckLevelAndCount(tr.subqueue, newAckLevel, -numAcked)
+	tr.lock.Unlock()
```

Comment:
There's a race causing the `should not move backwards ` assertion failure if the unlock happens before the call to `updateAckLevelAndCount`.